api/agent/drivers/docker/docker.go

```diff
@@ -53,16 +53,63 @@ func (r *runResult) Error() error   { return r.err }
 func (r *runResult) Status() string { return r.status }
 
 type DockerDriver struct {
-	conf     drivers.Config
-	docker   dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
-	hostname string
-	auths    map[string]driverAuthConfig
-	pool     DockerPool
+	conf       drivers.Config
+	docker     dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
+	hostname   string
+	auths      map[string]driverAuthConfig
+	pool       DockerPool
+	imageCache *Cache
 	// protects networks map
 	networksLock sync.Mutex
 	networks     map[string]uint64
 }
 
+func NewImageCleaner(dockerDriver *DockerDriver, context context.Context, maxSize int64) error {
+	opts := docker.RemoveImageOptions{}
+	duopts := docker.DiskUsageOptions{}
+	opts.Force = true
+	opts.NoPrune = false
+	opts.Context = context
+	ticker := time.NewTicker(dockerDriver.conf.ImageCacheCleanInterval)
+	for {
+		select {
+		case <-context.Done():
+			{
+				return nil
+			}
+		case <-ticker.C:
+			{
+				du, err := dockerDriver.docker.DiskUsage(duopts)
+				if err != nil {
+					logrus.WithError(err).Error("attempting to check disk usage")
+				}
+				if du.LayersSize > maxSize {
+					toEvict := dockerDriver.imageCache.Evictable()
+					for _, i := range toEvict {
```

Comment:
This is a simple for-loop. Evict list sorted or not doesn't matter, right?

Reply:
Some will probably fail. I want it to try the most likey to be useful evict first. As well as be able to abort the loop if enough space has been freed. On the other hand if its more complicated then it's worth we can totally delete it.